### PR TITLE
fix: UI part for logs RBAC - do not display the logs tab when no RBAC in place (#7211)

### PR DESCRIPTION
**Pull request by reggie-k**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [X] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [X] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by reggie-k** [comment]
Hope this one can be merged with squash commits.

**Comment by reggie-k** [comment]
@crenshaw-dev Thanks for changing the name :)
I am not sure about the docs though - the known UI issue note is relevant for those, who use <= 2.4.3.
Now, if this one makes it to 2.4.4, then 2.4.4 and above does not need the note, indeed.
The docs are on a minor version level, right? If I remove the note in this PR, the docs will be changed and <= 2.4.3 users will see neither the note nor the fixed behavior - right?

**Comment by reggie-k** [comment]
I will add it then.
If the commit eventually goes to 2.4.5 or later, I will change the doc to
contain the correct version, I will just need to know before it is merged 🙂

File: CHANGELOG.md
Comment (crenshaw-dev) [inline CHANGELOG.md:47]: Accidental copy/paste?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Review by rbreeze — COMMENTED** [review]
LGTM, once the accidental docs edit is fixed 🙂

**Comment by reggie-k** [comment]
I am a bit concerned about the fix because I only tested it manually. Tested thoroughly, of course, for both allow and deny use cases, and it behaves like I expect, but ArgoCD server side has many unit and end-to-end tests for multiple components, while the UI side has much less overall.
I am very new to React, was hoping to find some other UI pod view tests I can look at and then write my own based on them, but didn't find relevant ones.
I can learn more and write pod view tests, not only for this particular change, but wider, this will obviously take some time, what do you guys think about the PR from the release perspective? Goes in or waits for automatic tests?

**Comment by reggie-k** [comment]
Sure, I will update it. If you can test it as well it would be awesome!

**Comment by reggie-k** [comment]
Converted to draft since found some problems upon additional testing, still fixing

**Review by crenshaw-dev — APPROVED** [review]
Looks good on local test! Will hold off merging in case I'm right about the `Boolean` call.  :-)

File: ui/src/app/applications/components/resource-details/resource-details.tsx
Comment (reggie-k) [reply ui/src/app/applications/components/resource-details/resource-details.tsx]: Done

**Comment by crenshaw-dev** [comment]
@reggie-k apologies, I mean to get this into 2.4.6 and failed to do so. Can you push a docs update to say 2.4.7? Hopefully that push will also fix the hung DCO check. :-P

**Comment by crenshaw-dev** [comment]
Brilliant, thank you!

**Comment by crenshaw-dev** [comment]
I 100% forgot to cherry-pick this. Will add a commit to update docs.

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.9.

**Comment by reggie-k** [comment]
Great, thanks!

**Pull request merged**
